src/lib.rs

```diff
@@ -78,69 +80,73 @@ impl Agent {
+fn income_in_wneigh(s_network: SNetwork, adress: Address, dist: f32) -> Vec<i32> { // add walking distance to define walking neigh to global parameters
     // returns income distrib within walking distance
+    todo!()
 }
 
-fn activity_in_tneigh(t_network: TNetwork, adress: Address, time: f32) { // add transport time to define transport neigh to global parameters
+fn activity_in_tneigh(t_network: TNetwork, adress: Address, time: f32) -> Vec<i32> { // add transport time to define transport neigh to global parameters
     // returns number of each activity within transport distance
+    todo!()
 }
 
 fn mean(list: &[i32]) -> f64 {
     let sum: i32 = Iterator::sum(list.iter());
     f64::from(sum) / (list.len() as f64)
 }
 
-pub fn behaviour_walking(agent: Agent, rate: f32, house: House, s_network: SNetwork, dist: f32) {
-    let i = income(agent, rate);
+pub fn behaviour_walking(agent: &Agent, rate: f32, house: House, s_network: SNetwork, dist: f32) {
+    let i = agent.income(rate);
     let j = income_in_wneigh(s_network, agent.house.adress, dist);
     let jj = income_in_wneigh(s_network, house.adress, dist);
     match agent.behaviour_walk {
-        "Statusquo" => 1/(mean(j) - mean(jj)).abs(),
-        "MaxSim" => 1/(i - mean(jj)).abs(),
-        "KotH" => i - mean(jj),
-        "PoBA" => mean(jj) - i,
-        "Rooted" => 1,
+        BehaviourWalking::Statusquo => 1/(mean(j) - mean(jj)).abs(),
+        BehaviourWalking::MaxSim => 1/(i - mean(jj)).abs(),
+        BehaviourWalking::KotH => i - mean(jj),
+        BehaviourWalking::PoBA => mean(jj) - i,
+        BehaviourWalking::Rooted => 1,
     }
 }
 
 pub fn behaviour_transport(agent: Agent, rate: f32) {
     let (j,l,s) = activity_in_tneigh(s_network, agent.house.adress, dist);
     let (jj,ll,ss) = activity_in_tneigh(s_network, house.adress, dist);
     match agent.behaviour_trans {
-        "Statusquo" => 1/((j - jj).abs() + (l - ll).abs() + (s - ss).abs()),
-        "MaxOpp" => jj + ll + ss,
-        "Picky" => ll + ss - jj,
-        "None" => 1,
+        BehaviourTransport::Statusquo => 1/((j - jj).abs() + (l - ll).abs() + (s - ss).abs()),
+        BehaviourTransport::MaxOpp => jj + ll + ss,
+        BehaviourTransport::Picky => ll + ss - jj,
+        BehaviourTransport::None => 1,
     }
 }
 
 // ***** Affordability rating *****
 
-pub fn affordability(agent: Agent, house: House, rate: f32, price_to_income: f32, fund: f32) { // add price_to_income to global parameters
-    let i = price_to_income * income(agent, rate) + fund;
+pub fn affordability(agent: Agent, house: House, rate: f32, price_to_income: f32, fund: f32) -> f32 { // add price_to_income to global parameters
+    let i = price_to_income * agent.income(rate) + fund;
     let p = house.price;
     if p > i {
-        1 / (1 + ((p - 1.5 * i)/6).exp())
+        1.0 / (1.0 + ((p - 1.5 * i)/6.0).exp())
```

Comment:
Rust is picky about mixing integers and floats. It can be annoying sometimes. If you write a constant number, `3.0` makes it a float